src/tests/ftest/cart/dual_iface_server.c

```diff
@@ -575,6 +576,7 @@ int main(int argc, char **argv)
 	printf("----------------------------------------\n\n");
 
 	/* Spawn 2 servers, each one reads and writes URIs into diff file */
+	umask(600);
```

Comment:
this should be umask(S_IWGRP | S_IWOTH);  instead. umask param turns off bits set in permissions, so the current umask(600) will make it not read/writeable by the user

Reply:
Curious -- Is coverity saying the *default* `umask` is insecure?  Will coverity prefer we set `umask(S_IWGRP|S_IWOTH)`, even if it's more permissive than the default `umask`?  ([This thread](https://github.com/cockpit-project/cockpit/issues/760) claims the `SECURE_TEMP` warning is for old, insecure `mkstemp` implementations.)

Reply:
@mallove79 I am not really sure if it was complaining specifically about default permissions. The way I understood it the issue was no default permissions were being set on the files. The coverity message says this:
"An attacker may be able to guess the name of the temporary file, or the permissions may be too loose."
After looking into the mkstemp function it says you should always set umask before use because default file permissions can change depending on the platform, so I understood it like it wants you to explicitly set them. I saw that thread and that is why I initially set the umask to 0600, but I haven't really used this function before so I cannot say for sure what is the right approach here.

Reply:
for this instance it doesnt really matter, as it is not a production code and rather a standalone debug tool to help with connectivity reproduction issues.
in cart we recently made a change to umask due to coverity and we are also using  S_IWGRP | S_IWOTH.
So this should be fine here as well just to make coverity calm; the alternative would be to mark it as non-issue, but then we would have to keep doing it every time.